### ISSUE TITLE
Is there a way to scope the requests handled by xdomain?

### ISSUE THREAD

**Issue by kswin** [jpillora/xdomain#109]
Hello,

When loading xdomain, it looks like all xhrs are affected. I'm interested in targeting specific api calls with xdomain. I apologize if this isn't the correct place to post these questions.

Thanks

**Comment by jpillora** [comment]
Yes this is the right place for questions and yes xdomain hooks all xhr, if
the use case is you'd like some requests to use CORS, then the answer is:
just don't add a slave for those domains, or if it's the same domain, just
white list the desired path

**Comment by kswin** [comment]
so, if I only want the slave to process requests going to "my-api-target", then on the slave, I would have something like:

<script src="//cdn.rawgit.com/jpillora/xdomain/gh-pages/dist/0.6/xdomain.min.js" master="my-api-target"></script>

Not sure if I got that right... i really appreciate the help!

**Comment by jpillora** [comment]
Yep, given my-api-target is a domain

**Comment by kswin** [comment]
Actually, I think i need to elaborate my use-case. I changed the master attribute and it wasn't the outcome that I expected.

I have an app that makes xhr requests to the same domain and a different api domain. On my app, I have: 
    slaves = {};
    slaves[apiDomain] = '/proxy.html';
    xdomain.slaves(slaves);

From the api domain, I serve proxy.html with:
    <script src="//cdn.rawgit.com/jpillora/xdomain/gh-pages/dist/0.6/xdomain.min.js" data-master="*"></script> 

This caused all xhr calls from my app to go through xdomain, including a $.ajax call to the app domain with async=false. xdomain doesn't support sync requests so it caused an error in the app. This illustrated that we wanted to scope xdomain to requests that matched the apiDomain. I tried to change data-master from \* to the apiDomain, but now the cross domain call doesn't work anymore. The Networks tab on dev tools does not display any calls to apiDomain and the console reads "xdomain (apiDomain): blocked request from: 'app domain'. Also, I still see the error caused by $.ajax async=false.

I would like calls going to the apiDomain to be processed by xdomain, but only those. Does the additional background give any ideas on what I'm doing wrong? Or maybe it's inherent in the nature of xdomain that all xhr requests will be processed by it?
